src/base/http/connection.h

```diff
@@ -27,41 +28,69 @@
  * exception statement from your version.
  */
 
-
 #pragma once
 
 #include <QElapsedTimer>
 #include <QObject>
 
+#include "types.h"
+
 class QTcpSocket;
 
 namespace Http
 {
-    class IRequestHandler;
-    struct Response;
-
     class Connection : public QObject
     {
         Q_OBJECT
         Q_DISABLE_COPY(Connection)
 
     public:
-        Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent = nullptr);
-        ~Connection();
+        enum Status
+        {
+            PendingRequest,
+            PendingStatus,
+            PendingHeaders,
+            PendingContent,
+            WritingContent,
+            Closed
+        };
+
+        // Connection takes ownership of socket
+        explicit Connection(QTcpSocket *socket, QObject *parent = nullptr);
 
-        bool hasExpired(qint64 timeout) const;
-        bool isClosed() const;
+        void sendStatus(const ResponseStatus &status);
+        void sendHeaders(const HeaderMap &headers);
+        void sendContent(const QByteArray &data);
+        void close();
+
+        Http::Environment enviroment() const;
+        Status status() const;
+        qint64 inactivityTime() const;
+        qint64 bytesToWrite() const;
+
+    signals:
+        void bytesWritten(qint64 bytes);
+        void disconnected();
+        void readyRequest(Request request);
 
     private slots:
         void read();
 
     private:
-        static bool acceptsGzipEncoding(QString codings);
-        void sendResponse(const Response &response) const;
+        enum ConnectionType
+        {
+            Unknown,
```

Comment:
Is this "pseudo" type required anywhere in the code? If it's only to initialize `m_connectionType` you can easily drop it and use any of real type.

Reply:
@jagannatharjun 
Don't invent states by yourself. Read the references if its usage unclear to you.

Reply:
I added it to denote the unknown state when we didn't received the headers yet. I don't think it's used anywhere though

Reply:
You can use any of valid types by default and then change it with the one obtained from header. Anyway it shouldn't be used until header is received.